ansible_builder/introspect.py

```diff
@@ -5,15 +5,31 @@
 import yaml
 import argparse
 
+
 base_collections_path = '/usr/share/ansible/collections'
 default_file = 'execution-environment.yml'
 
 
+def pip_file_data(path):
+    req_list = []
+    with open(path, 'r') as f:
+        for line in f.read().split('\n'):
+            if not line:
+                continue
+            if line.startswith('-r') or line.startswith('--requirement'):
+                _, new_filename = line.split(None, 1)
+                new_path = os.path.join(os.path.dirname(path or '.'), new_filename)
+                req_list.extend(pip_file_data(new_path))
+            else:
+                req_list.append(line)
+    return req_list
```

Comment:
My first attempt looked nothing like this:

https://stackoverflow.com/questions/49689880/proper-way-to-parse-requirements-file-after-pip-upgrade-to-pip-10-x-x/59969843#comment111716393_59969843

I found importing from `pip` to be pretty unusable. Maybe there's something I didn't get, but...

https://github.com/davidfischer/requirements-parser/blob/ad404add0423d7e1f419c3a48dc884375d57f376/requirements/parser.py#L33-L39

This is how the `requirements-parser` library handles these cross-file cases. It seemed relatively uncomplicated, and there's not much wiggle room in the allowed syntax to put in more bizarre forms of references.